tests/training_success/explore_schema.py

```diff
@@ -0,0 +1,422 @@
+
+    This function does not take a cross product of all the options for all the config parameters. It selects parameter
+    values independently from each other.
+
+    Args:
+        config_options: dictionary mapping from ludwig config parameter to all values to be explored.
+            Here's an example of what it could look like:
+
+                trainer.batch_size: ["auto", 2, 43]
+                trainer.learning_rate: ["auto", 0.1, 0.00002, 0.32424]
+                ...
+    """
+    num_configs = 1
+    for key in config_options:
+        if isinstance(config_options[key], list):
+            num_configs = max(num_configs, len(config_options[key]))
+            config_options[key] = deque(config_options[key])
+
+    for _ in range(num_configs):
+        config = {}
+        for key in config_options:
+            if config_options[key] and not isinstance(config_options[key], str):
+                config[key] = config_options[key].popleft()
+            elif isinstance(config_options[key], str):
+                config[key] = config_options[key]
+        yield config
+
+
+def create_nested_dict(flat_dict: Dict[str, Union[float, str]]) -> Dict[str, Any]:
+    """Generate a nested dict out of a flat dict whose keys are delimited by a delimiter character.
+
+    Params:
+        flat_dict: potential generated baseline config. Here's an example of what it could look like:
+
+            trainer.batch_size: 324
+            trainer.learning_rate: 0.0635
+
+        The expected output would be
+
+            trainer:
+                batch_size: 324
+                learning_rate: 0.0635
+    """
+
+    def to_nested_format(key: str, value: Union[str, int, float], delimiter: str = ".") -> Dict[str, Any]:
+        # https://stackoverflow.com/a/40401961
+        split_key = key.split(delimiter)
+        for key in reversed(split_key):
+            value = {key: value}
+        return value
+
+    config = {}
+    for key in flat_dict:
+        config = merge_dict(config, to_nested_format(key, copy.deepcopy(flat_dict[key])))
+    return config
+
+
+def combine_configs(
+    explored: Deque[Tuple[Dict, bool]], config: ModelConfigDict
+) -> List[Tuple[ModelConfigDict, pd.DataFrame]]:
+    """Merge base config with explored sections.
+
+    Args:
+        explored: deque containing all the config options.
+        config: base Ludwig config to merge the explored configs with.
+    """
+    dataset = build_synthetic_dataset_df(10, config)
+    ret = []
+    for config_options, _ in explored:
+        for default_config in generate_possible_configs(config_options=config_options):
+            default_config = create_nested_dict(default_config)
+            merged_config = merge_dict(copy.deepcopy(config), default_config)
+            try:
+                ModelConfig.from_dict(merged_config)
+                ret.append((merged_config, dataset))
+            except Exception:
+                pass
```

Comment:
It could be potentially useful to return a count of the number of Excepted configs.

As part of the combinatorial tests, we could check that the number of excepted configs is below some cap and if the cap were to be exceeded significantly, that would indicate that there's either been a bad schema change, or that the combinatorial sampling might warrant some improving.